src/encoded/static/components/item-pages/MicroscopeConfigurationView.js

```diff
@@ -0,0 +1,412 @@
+                <i className={"icon icon-fw icon-" + (cloneLoading ? 'circle-notch icon-spin fas' : 'clone far')}/>&nbsp; Clone
+            </button>
+        );
+    }
+
+    /**
+     * Is used to call {function} `props.toggleFullScreen` which is passed down from app.js BodyElement.
+     */
+    handleFullscreenToggle(){
+        const { isFullscreen, toggleFullScreen } = this.props;
+        setTimeout(toggleFullScreen, 0, !isFullscreen);
+    }
+
+    fullscreenButton(){
+        const { isFullscreen, toggleFullScreen } = this.props;
+        if(typeof isFullscreen === 'boolean' && typeof toggleFullScreen === 'function'){
+            return (
+                <button type="button" className="btn btn-outline-dark" onClick={this.handleFullscreenToggle} data-tip={!isFullscreen ? 'Expand to full screen' : null}>
+                    <i className={"icon icon-fw fas icon-" + (!isFullscreen ? 'expand' : 'compress')}/>
+                </button>
+            );
+        }
+        return null;
+    }
+
+    onSaveMicroscope(microscope, complete) {
+        // // Do some stuff... show pane for people to browse/select schema.. etc.
+        // setTimeout(function() {
+        //     console.log(microscope);
+        //     complete(microscope.Name);
+        // });
+
+        const { context, href } = this.props;
+        // We're updating this object's microscope_settings.
+        const payload = { 'microscope_settings': microscope };
+        console.log(microscope);
+
+        this.setState({ 'saveLoading': true, 'modal': null }, () => {
+            ajax.load(
+                href,
+                (resp) => {
+                    // Success callback... maybe update state.originalViewConfigString or something...
+                    // At this point we're saved maybe just notify user somehow if UI update re: state.saveLoading not enough.
+                    Alerts.queue({
+                        'title': "Saved " + microscope.Name,
+                        'message': "This Microscope Configuration Item has been updated with the current viewport. This may take a few minutes to take effect.",
+                        'style': 'success'
+                    });
+                    this.setState({ 'saveLoading': false });
+                },
+                'PATCH',
+                () => {
+                    // Error callback
+                    Alerts.queue({
+                        'title': "Failed to save display.",
+                        'message': "Sorry, can you try to save again?",
+                        'style': 'danger'
+                    });
+                    this.setState({ 'saveLoading': false });
+                },
+                JSON.stringify(payload)
+            );
+        });
+        complete(microscope.Name);
+    }
+
+    render(){
+        const { schemas, context, windowWidth, windowHeight } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['schemas' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4271272)